smst.go

```diff
@@ -150,11 +172,39 @@ func (smst *SMST) Root() MerkleRoot {
 // Sum returns the sum of the entire trie stored in the root.
 // If the tree is not a sum tree, it will panic.
 func (smst *SMST) Sum() uint64 {
-	rootDigest := smst.Root()
+	rootDigest := []byte(smst.Root())
+
 	if !smst.Spec().sumTrie {
 		panic("SMST: not a merkle sum trie")
 	}
+
+	firstSumByteIdx, firstCountByteIdx := GetFirstMetaByteIdx(rootDigest)
+
 	var sumBz [sumSizeBytes]byte
-	copy(sumBz[:], []byte(rootDigest)[len([]byte(rootDigest))-sumSizeBytes:])
+	copy(sumBz[:], rootDigest[firstSumByteIdx:firstCountByteIdx])
 	return binary.BigEndian.Uint64(sumBz[:])
 }
+
+// Count returns the number of non-empty nodes in the entire trie stored in the root.
+func (smst *SMST) Count() uint64 {
+	rootDigest := []byte(smst.Root())
+
+	if !smst.Spec().sumTrie {
+		panic("SMST: not a merkle sum trie")
+	}
+
+	_, firstCountByteIdx := GetFirstMetaByteIdx(rootDigest)
+
+	var countBz [countSizeBytes]byte
+	copy(countBz[:], rootDigest[firstCountByteIdx:])
+	return binary.BigEndian.Uint64(countBz[:])
+}
+
+// GetFirstByte returns the index of the first count byte and the first sum byte
+// in the data slice provided. This is useful metadata when parsing the data
+// of any node in the trie.
+func GetFirstMetaByteIdx(data []byte) (firstSumByteIdx, firstCountByteIdx int) {
```

Comment:
```suggestion
func getFirstMetaByteIdx(data []byte) (firstSumByteIdx, firstCountByteIdx int) {
```
IMO this should be private